src/main/java/core/basesyntax/service/operations/PurchaseOperationsHandler.java

```diff
@@ -0,0 +1,16 @@
+package core.basesyntax.service.operations;
+
+import core.basesyntax.model.Operations;
+
+public class PurchaseOperationsHandler implements OperationsHandler {
+
+    @Override
+    public Operations getOperation() {
+        return Operations.PURCHASE;
+    }
+
+    @Override
+    public int getOperationsResult(int result, int quantity) {
+        return result - quantity;
```

Comment:
When handling the Purchase operation, you need to check if the result balance is positive before saving it in the Storage. If the balance after purchase is negative, you should throw a RuntimeException.